Model/Behavior/AuditableBehavior.php

```diff
@@ -366,17 +379,52 @@ protected function _getModelData(Model $Model) {
 
 		foreach ($this->settings[$Model->alias]['habtm'] as $habtmModel) {
 			if (array_key_exists($habtmModel, $Model->hasAndBelongsToMany) && isset($data[$habtmModel])) {
-				$habtmIds = Hash::combine(
-					$data[$habtmModel],
-					'{n}.id',
-					'{n}.id'
-				);
+				if (isset($this->settings[$Model->alias]['deepLog']) && $this->settings[$Model->alias]['deepLog']) {
```

Comment:
Can we extract that part into a protected method, something with "deepLog"? Methods are already way too long.